desktop/flipper-ui-core/src/dispatcher/pluginMarketplace.tsx

```diff
@@ -0,0 +1,209 @@
+  return selectCompatibleMarketplaceVersions(availablePlugins);
+}
+
+async function refreshMarketplacePlugins(store: Store): Promise<void> {
+  if (getFlipperLib().isFB && !isLoggedIn().get()) {
+    // inside FB we cannot refresh when user is not logged
+    return;
+  }
+  try {
+    const plugins = await loadPluginsFromMarketplace();
+    store.dispatch(registerMarketplacePlugins(plugins));
+    autoUpdatePlugins(store, plugins);
+  } catch (err) {
+    if (isConnectivityOrAuthError(err)) {
+      // This is handled elsewhere and we don't need to create another warning or error for it.
+      console.warn(
+        'Connectivity or auth error while refreshing plugins from Marketplace',
+        err,
+      );
+    } else {
+      console.error('Error while refreshing plugins from Marketplace', err);
+    }
+  }
+}
+
+export function autoUpdatePlugins(
+  store: Store,
+  marketplacePlugins: DownloadablePluginDetails[],
+) {
+  const state = store.getState();
+  if (!state.plugins.initialized) {
+    // skip auto-updating plugins if they are not initialized yet
+    return;
+  }
+  const {loadedPlugins, installedPlugins, uninstalledPluginNames} =
+    state.plugins;
+  const autoInstalledPlugins = new Set<string>();
+  for (const client of state.connections.clients.values()) {
+    const enabledPlugins = state.connections.enabledPlugins[client.query.app];
+    if (enabledPlugins) {
+      // If we already have persisted list of enabled plugins -
+      // we should install those of them which are enabled, but not installed by some reason.
+      enabledPlugins.forEach((p) => {
+        if (client.supportsPlugin(p)) {
+          autoInstalledPlugins.add(p);
+        }
+      });
+    } else {
+      // If there is no persisted list of enabled plugins this means that user
+      // opened this app for first time. In such case we should enable and install
+      // plugins which are enabled by default.
+      for (const plugin of marketplacePlugins) {
+        if (plugin.isEnabledByDefault && client.supportsPlugin(plugin.id)) {
+          autoInstalledPlugins.add(plugin.id);
+          const loadedPluginInstance = state.plugins.clientPlugins.get(
+            plugin.id,
+          );
+          if (loadedPluginInstance) {
+            // If plugin was already installed before (e.g. for debugging another mobile app),
+            // then we should switch its state to "loaded" for the current app.
+            store.dispatch(
+              switchPlugin({
+                plugin: loadedPluginInstance,
+                selectedApp: client.query.app,
+              }),
+            );
+          } else {
+            // If plugin was not installed before, then we should mark it as enabled
+            // to ensure it is automatically loaded after downloaded from Marketplace.
+            store.dispatch(setPluginEnabled(plugin.id, client.query.app));
+          }
+        }
+      }
+    }
+  }
+  if (isAutoUpdateDisabled() || !isLoggedIn().get()) {
```

Comment:
This condition will never allow for auto update to happen, since the `isLoggedIn().get()` returns `false`.

So either we can rewrite the stub for OS, or remove those `isLoggedIn` conditions from OS code.

```js
const isLoggedInAtom = createState(false);
export function isLoggedIn(): Atom<boolean> {
  return isLoggedInAtom;
}
```

Reply:
Changing stub to always return true in OS sounds good to me